packages/govuk-frontend/src/govuk/common/index.mjs

```diff
@@ -134,21 +134,33 @@ export function getFragmentFromUrl(url) {
 }
 
 /**
- * Checks if GOV.UK Frontend is supported on this page
- *
- * Some browsers will load and run our JavaScript but GOV.UK Frontend
- * won't be supported.
  *
  * @internal
  * @param {HTMLElement | null} [$scope] - HTML element `<body>` checked for browser support
- * @returns {boolean} Whether GOV.UK Frontend is supported on this page
+ * @returns {string} Support level
  */
-export function isSupported($scope = document.body) {
-  if (!$scope) {
-    return false
+export function getSupportedLevelMessage($scope = document.body) {
+  const LEVELS = {
+    OK: 'OK',
+    NO_SUPPORT: 'GOV.UK Frontend is not supported in this browser',
+    NO_MODULE: 'GOV.UK Frontend initialised without `<script type="module">`',
+    MISSING_CLASS: '<body> tag is missing the `govuk-frontend-supported` class'
+  }
+  let supportedLevel = LEVELS.OK
+
+  // The currentScript property only exists on <script> tags with no module.
+  // $scope will be null if `initAll` was called from the <head> with no module
+  if (document.currentScript ?? !$scope) {
```

Comment:
I think this means that GOV.UK Frontend will now 'refuse to run' if it's loaded without using `type="module"`?

That might be desired, but we should make that change consciously.

Reply:
Older browsers that _could_ potentially see "**GOV.UK Frontend is not supported in this browser**" will probably crash on ES2015+ syntax anyway and won't reach the `isSupported()` check 😭

(Other than Safari 10.1)

But I love the idea of also running the `!('noModule' in HTMLScriptElement.prototype)` check in our JavaScript so we can swap out the error message for services that skip the new snippet

Could we just do the error message switching in **src/govuk/errors/index.mjs**?

```mjs
const supportMessage =
  'noModule' in HTMLScriptElement.prototype
    ? 'GOV.UK Frontend is missing `<body class="govuk-frontend-supported">` via inline `<script>`'
    : 'GOV.UK Frontend is not supported in this browser'

super(
  $scope
    ? supportMessage
    : 'GOV.UK Frontend initialised without `<script type="module">`'
)
```

Reply:
It's a great point - think this would mess with folks who use UMD, for example?